src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php

```diff
@@ -174,28 +184,42 @@ public function all()
             $log = [];
 
             foreach ($lines as $key => $line) {
-                $log[] = [
-                    'context' => '',
-                    'level' => '',
-                    'level_class' => '',
-                    'level_img' => '',
-                    'date' => $key + 1,
-                    'text' => $line,
-                    'in_file' => null,
-                    'stack' => '',
-                ];
+                $log[] = $this->getArrayLog('', '', '', $key, $line);
             }
         }
 
         return array_reverse($log);
     }
 
+    /**
+     * Create array data from log
+     * @param $index
+     * @param $current
+     * @param $level
+     * @param $key
+     * @param $line
+     * @return array
+     */
+    protected function getArrayLog($index, $current, $level, $key, $line)
```

Comment:
I'm not totally sure this function makes the code more readable (but maybe I'm wrong). I mean it adds a bunch of ternary operators (7 ternary operator) and is more complex to read. It avoid repetition of keys, good point, but it adds some complexity IMO. And there is a new function with _5_ params which are not really simple to understand TBH. I don't _actually_ think it's more readable, but I understand your idea. What are your thoughts?

Reply:
What about this approach
```php
$log[] = $this->getArrayLog([
                    'index' => '',
                    'current' => [],
                    'level' => '',
                    'key' => $key,
                    'line' => $line
                ]);
```
And the function 
```php
/**
     * Create array data from log
     * @param $data
     * @return array
     */
    protected function getArrayLog($data)
    {
        $index = $data['index'];
        $current = $data['current'];
        $level = $data['level'];
        $key = $data['key'];
        $line = $data['line'];
```